Use HappyPack and HardSource Plugin to make builds fast

sloanelybutsurely opened the pull request:
This speeds up initial builds quite a bit. It was able to build all of `member_profile_app` in 10s in dev mode. 

This is something I'd like feedback on before merging. It adds a bit of indirection to the configuration of the loaders so there's a trade off there. Also, would we want to do the same thing in the production build?

[HappyPack](https://github.com/amireh/happypack)

~~I'm going to see if adding the [`hard-source-webpack-plugin`](https://github.com/mzgoddard/hard-source-webpack-plugin) makes it any faster also.~~ This makes the second time you do an initial build _insanely_ fast! 🚄  🚀  🎸 
@trunkclub/ui

Commit:
Use HappyPack to run loaders in parallel

Commit:
Make initial builds BLAZING fast w/ hard-source-webpack-plugin

Commit:
Add `environmentPaths` to bust cache

Commit:
Package deps for experimental release

zperrault commented:
Not sure where the `.happypack` and `.cache` directories should go... 😕

Commit:
Move caches into ./node_modules/.cache

Commit:
Change release script

kellyrmilligan commented:
@trunkclub/ui just to baseline this change, this could also benefit the other proposal as well since it is contained in webpack. I would be happy to explore optimizations in webpack once the group has decided on which direct we are taking things. this change would also benefit the current @trunkclub/build process.

zperrault commented:
Currently working on getting this to work with the bundled deps. `hard-source-webpack-plugin` has some deps that don't like being bundled. Going to try to allow them to be installed normally.

Commit:
Merge branch 'master' of github.com:trunkclub/create-react-app into run-in-parallel

Commit:
Remove hard-source-webpack-plugin from bundle

This package does not work when bundled. There are several postinstall
scripts that cause installation to fail when it is bundled.

Commit:
Ready to cut release
